### PR TITLE
Basilisk II 68k Macintosh emulator

### PR DESCRIPTION
**Pull request by SegHaxx**
Hi this is a flatpak for the Basilisk II 68k Macintosh emulator. This is my first ever flatpak so apologies in advance if I did anything horribly wrong. :)

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.haxxed.BasiliskII

**Comment by SegHaxx** [comment]
Oh I see I made a silly error. :)

File: BasiliskII.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (bilelmoussaoui) [inline BasiliskII.desktop:1]: Should be renamed to match the app-id & contributed upstream if possible

File: com.haxxed.BasiliskII.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+app-id: com.haxxed.BasiliskII
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: BasiliskII
+finish-args:
+  - --device=dri
+  - --socket=pulseaudio
+  - --socket=wayland
```
Comment (bilelmoussaoui) [inline com.haxxed.BasiliskII.yml:9]: is it wayland compatible?

File: BasiliskII.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>com.haxxed.BasiliskII</id>
```
Comment (bilelmoussaoui) [inline BasiliskII.metainfo.xml:3]: Should be upstreamed as well

Comment (SegHaxx) [reply BasiliskII.metainfo.xml:3]: That's the plan, yes. Just wanted to make sure it's acceptable before I merge it.

File: com.haxxed.BasiliskII.yml (hunk 2 above)
Comment (SegHaxx) [reply com.haxxed.BasiliskII.yml:9]: I assume so? It doesn't use X directly, the app uses GTK2 and SDL2? So it's up to what they do. (doesn't look like GTK2 does wayland)

File: BasiliskII.desktop (hunk 1 above)
Comment (SegHaxx) [reply BasiliskII.desktop:1]: Can do.

**Comment by barthalion** [comment]
bot, build com.haxxed.BasiliskII

(FYI, you can start new builds yourself.)

**Comment by barthalion** [comment]
bot, build com.haxxed.BasiliskII

File: BasiliskII.metainfo.xml (hunk 3 above)
Comment (barthalion) [inline BasiliskII.metainfo.xml:3]: This ID is certainly wrong. If the website is https://www.emaculation.com/doku.php/basilisk_ii, it should me `com.emaculation.basilisk_ii` (or something along this). In general it should match a domain used by the project.

File: com.haxxed.BasiliskII.yml (hunk 2 above)
Comment (barthalion) [reply com.haxxed.BasiliskII.yml:9]: Yeah, it doesn't work on Wayland (maybe SDL2 part does but not the launcher).

File: com.haxxed.BasiliskII.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  - --socket=x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=home
```
Comment (barthalion) [inline com.haxxed.BasiliskII.yml:13]: Does it require full home access or can it be limited to specific dirs or at least read-only?

File: BasiliskII.metainfo.xml (hunk 3 above)
Comment (barthalion) [reply BasiliskII.metainfo.xml:3]: No, it's fine to handle it later (either after merge or still during review).

Comment (SegHaxx) [reply BasiliskII.metainfo.xml:3]: Well that's kind of the problem with this project. It is very old, the old project leadership has mostly disappeared, so I'm part of a small band of users trying to revive it to modern standards, loosely based around the emaculation forums. I can certainly see about merging this all into the emaculation/macemu project if required.

Comment (barthalion) [reply BasiliskII.metainfo.xml:3]: It should be a reverse DNS of an "owned" domain, so if it's mostly a personal fork, `com.github.username.reponame` is also good.

File: com.haxxed.BasiliskII.yml (hunk 4 above)
Comment (SegHaxx) [reply com.haxxed.BasiliskII.yml:13]: It read/writes VM disk images and reads ROM files that could potentially be anywhere. It also currently uses old school ~/.basilisk* files for preferences and state. For this to be a drop-in replacement for an existing Basilisk II user, home access would seem to be minimum. The user selects files once in the GUI then they are remembered so I don't know how portals work out with that use case. Also does the portal stuff work with GTK2? Porting to GTK3 is possible but that's a whole lot of work. I was hoping to get Basilisk II back in the hands of users, which would justify spending further time rewriting large portions of it. :(

As it is I keep my vintage Mac stuff on shares mounted in /srv or /run/user/*/gvfs so i'm already having to set overrides for my use case.

I'm open to suggestions how to handle this better. :)

File: BasiliskII.metainfo.xml (hunk 3 above)
Comment (ianfixes) [reply BasiliskII.metainfo.xml:3]: Hello, `emaculation/macemu` maintainer here.  I forked `macemu` from `cebix/macemu` in early 2019 when it appeared that the project was without leadership -- [only 2 merges between Nov 2018 and March 2020](https://github.com/cebix/macemu/pulls?page=1&q=is%3Apr+is%3Aclosed), and (at that time) only the rarely-seen owner had write access.  My short term intention for my fork was to revive the project if at all possible by setting up CI/CD and cherry picking some of the unmerged contributions to the `cebix` repo; my long term intention was to transfer ownership to one or more of those long-time contributors in the absence of the original maintainer.

Instead, within the past year the upstream `cebix/macemu` project seems to have passed the torch to a different maintainer (or maintainers) and resumed development.  Unless & until that team decides that they prefer my namespace to theirs, `cebix/macemu` should be considered the appropriate "upstream" for this package.

File: com.haxxed.BasiliskII.yml (hunk 4 above)
Comment (barthalion) [reply com.haxxed.BasiliskII.yml:13]: Yeah, I was just double-checking I'm not missing anything.

File: BasiliskII.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+Basilisk II is an Open Source 68k Macintosh emulator. That is, it enables you to run 68k Mac OS software on you computer, even if you are using a different operating system. However, you still need a copy of Mac OS 6.x-8.1 and a Macintosh ROM image to use Basilisk II.
+  </p></description>
+
+  <url type="homepage">https://www.emaculation.com/doku.php/basilisk_ii</url>
+
+  <launchable type="desktop-id">com.haxxed.BasiliskII.desktop</launchable>
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://pbs.twimg.com/media/En4E_C2VkAAojiG?format=png&amp;name=4096x4096</image>
```
Comment (AsciiWolf) [inline BasiliskII.metainfo.xml:22]: Screenshot url seems to return an empty page.

Comment (soredake) [reply BasiliskII.metainfo.xml:22]: correct url is https://pbs.twimg.com/media/En4E_C2VkAAojiG?format=png&name=4096x4096

File: BasiliskII.metainfo.xml (hunk 3 above)
Comment (SegHaxx) [reply BasiliskII.metainfo.xml:3]: Okay I'll set the app id to the cebix github.

Comment (SegHaxx) [reply BasiliskII.metainfo.xml:3]: Alright. What should the app id be then? net.cebix.basilisk?

File: com.haxxed.BasiliskII.yml (hunk 2 above)
Comment (SegHaxx) [reply com.haxxed.BasiliskII.yml:9]: Should I be removing the wayland perms then?

File: BasiliskII.metainfo.xml (hunk 5 above)
Comment (SegHaxx) [reply BasiliskII.metainfo.xml:22]: Build fails if I change this. TBH it's a placeholder anyway.

`
Error loading AppData file: failed to parse /app/share/appdata/net.cebix.basilisk.appdata.xml: Error on line 22: Entity did not end with a semicolon; most likely you used an ampersand character without intending to start an entity — escape ampersand as &amp;
`

**Comment by SegHaxx** [comment]
bot, build net.cebix.basilisk

**Comment by barthalion** [comment]
bot, build net.cebix.basilisk

**Comment by ianfixes** [comment]
@barthalion as [mentioned in a comment above](https://github.com/flathub/flathub/pull/1986#discussion_r540695090) I would want some guidance from the maintainers of `cebix/macemu` before pursuing anything here.  I maintain an experimental fork that has diverged from the main project and core development community.

**Pull request closed without merging**